File: packages/core/src/memory/MemoryItem.ts
import { createId } from "../utils/createId";

export interface MemoryItem {
  id: string;
  userId: string;
  content: string;
  tags: string[];
  importance: number;
  createdAt: string;
  updatedAt: string;
}

export function createMemoryItem(
  userId: string,
  content: string,
  tags: string[] = [],
  importance: number = 0.5
): MemoryItem {
  const now = new Date().toISOString();
  return {
    id: createId("memory"),
    userId,
    content,
    tags,
    importance,
    createdAt: now,
    updatedAt: now,
  };
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/memory/MemoryItem.kt
package ai.mobileagentos.compose.memory

import ai.mobileagentos.compose.utils.IdGenerator

data class MemoryItem(
    val id: String,
    val userId: String,
    val content: String,
    val tags: List<String>,
    val importance: Float,
    val timestamp: String
) {
    companion object {
        fun create(
            userId: String,
            content: String,
            tags: List<String> = emptyList(),
            importance: Float = 0.5f
        ): MemoryItem {
            return MemoryItem(
                id = IdGenerator.generate("memory"),
                userId = userId,
                content = content,
                tags = tags,
                importance = importance.coerceIn(0f, 1f),
                timestamp = System.currentTimeMillis().toString()
            )
        }
    }
}